### PR TITLE
universal: New package

### PR DESCRIPTION
**Pull request by eschnett**
Universal: a header-only C++ template library for universal number arithmetic

### CODE REVIEW

File: var/spack/repos/builtin/packages/universal/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class Universal(CMakePackage):
+    """Universal: a header-only C++ template library for universal number arithmetic"""
+
+    homepage = "https://github.com/stillwater-sc/universal"
+    url = "https://github.com/stillwater-sc/universal"
+
+    maintainers("eschnett")
+
+    # The release file "3.58b" contains version 3.59.1
+    version("3.59.1", sha256="771d64cf51862fdc7c54de585eafac67",
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/universal/package.py]: This is not the correct checksum.  I'm getting the following:  

```suggestion
    version("3.59.1", sha256="0d9346b828af4ac1af2eabceaba20ba4c1208ce8afa9a2a7bdba3a382555f700",
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by tldahlgren** [comment]
Why not just use the `3.59` tagged branch (https://github.com/stillwater-sc/universal/releases/tag/v3.59)?  Or have the version number match the version?

There are also newer versions (`3.68` as of yesterday) posted at https://github.com/stillwater-sc/universal/tags.

**Comment by eschnett** [comment]
The version with `3.58b` in the file name self-identifies as `3.59.1` in the source code. Since people will never see the file name I thought that the version number in the source code is more appropriate. I'll update to the newer version.

**Review by tldahlgren — APPROVED** [review]
Confirmed the sha256, rest LGTM

**Pull request merged**